marconi-chain-index/src/Marconi/ChainIndex/Error.hs

```diff
@@ -37,10 +36,22 @@ liftSQLError errorWrapper =
                 `catch` \(err2 :: SQL.ResultError) -> pure (Left (InvalidIndexer $ Text.pack $ show err2))
     )
 
-raiseException
-  :: ExceptT IndexerError IO a -> IO a
+raiseException :: (Exception err) => ExceptT (IndexerError err) IO a -> IO a
 raiseException x = do
   x' <- runExceptT x
   case x' of
     Left err -> throw err
     Right result -> pure result
+
+ignoreQueryError :: ExceptT (IndexerError err) IO a -> ExceptT (IndexerError Void) IO a
```

Comment:
By reading the code, it is not 100% clear to me why we need to use `Void`. Just for my own comprehension.

Reply:
Here is the rationale: each query has its own exception. Each indexer has one and one only `StorableMonad`, that must be able to handle these exceptions and the other indexer exceptions. To allow a specialisation by indexer, we provide a parametric `IndexerError` type.
But, when we deal with `Workers` in the `Indexers` type class, all indexers must have the same monad stack to manage block indexing in the same way. As during indexing query error can't occur, the easiest solution for this common monad stack is to claim that it can't occur, using `Void`.
So `ignoreQueryError` deals with the `QueryError` case but in practice, this case shouldn't appear.

Reply:
Thanks :)